library/alloc/src/vec/mod.rs

```diff
@@ -416,6 +416,7 @@ impl<T> Vec<T> {
     /// ```
     #[inline]
     #[rustc_const_stable(feature = "const_vec_new", since = "1.39.0")]
+    #[cfg_attr(not(test), rustc_diagnostic_item = "vec_new")]
```

Comment:
I mentioned on Zulip already, but we typically do this by making the item a diagnostic item then checking the method's identifier. That's typically how it's suggested to be done. But regardless this way is fine, and it doesn't need to be blocked on it

Reply:
Yeah, I had already implemented all this before talking on Zulip and seeing the idea to check method identifiers. Since there are already diagnostic items on methods, you are correct that this very much shouldn't be blocked on that though.